extensions/python/pythonlibloader/PythonLibLoader.cpp

```diff
@@ -0,0 +1,96 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#if !defined(WIN32) && !defined(__APPLE__)
+#include <dlfcn.h>
+#include <cstdio>
+#include <iostream>
+#include <string>
+#include <array>
+#include "utils/StringUtils.h"
+#include "core/logging/LoggerConfiguration.h"
+#endif
+#include "core/extension/Extension.h"
+
+namespace minifi = org::apache::nifi::minifi;
+
+#if !defined(WIN32) && !defined(__APPLE__)
+class PythonLibLoader {
+ public:
+  explicit PythonLibLoader(const std::shared_ptr<minifi::Configure>& config) {
+    std::string python_command = "python3";
+    if (auto python_binary = config->get(minifi::Configure::nifi_python_env_setup_binary)) {
+      python_command = python_binary.value();
+    }
```

Comment:
I don't think this finds the latest installed version e.g. on RHEL8, where python 3.6 and python 3.12 can be installed at the same time.

Reply:
We discussed this last week that we don't want to find the latest version instead it's preferable to find the library version associated with the binary set in the minifi.properties or the library associated with the default python3 command if the property is not set.